lib/IDE/Refactoring.cpp

```diff
@@ -3887,7 +3889,1442 @@ bool RefactoringActionConvertToComputedProperty::performChange() {
+        return addCustom(CE->getStartLoc(), CE->getEndLoc().getAdvancedLoc(1),
+                         [&]() { addHandlerCall(CE); });
+
+      auto *CE = dyn_cast_or_null<CallExpr>(E);
+      auto HandlerDesc = AsyncHandlerDesc::find(CE, /*ignoreName=*/true);
+      if (HandlerDesc.isValid())
+        return addCustom(CE->getStartLoc(), CE->getEndLoc().getAdvancedLoc(1),
+                         [&]() { addAsyncAlternativeCall(CE, HandlerDesc); });
+    }
+
+    return true;
+  }
+#undef PLACEHOLDER_START
+#undef PLACEHOLDER_END
+
+  bool walkToExprPost(Expr *E) override {
+    NestedExprCount--;
+    return true;
+  }
+
+  bool addCustom(SourceLoc End, SourceLoc NextAddedLoc,
+                 std::function<void()> Custom = {}) {
+    addRange(LastAddedLoc, End);
+    Custom();
+    LastAddedLoc = NextAddedLoc;
+    NestedExprCount--;
+    return false;
+  }
+
+  void addRange(SourceLoc Start, SourceLoc End, bool ToEndOfToken = false) {
+    if (ToEndOfToken) {
+      OS << Lexer::getCharSourceRangeFromSourceRange(SM,
+                                                     SourceRange(Start, End))
+                .str();
+    } else {
+      OS << CharSourceRange(SM, Start, End).str();
+    }
+  }
+
+  void addRange(SourceRange Range, bool ToEndOfToken = false) {
+    addRange(Range.Start, Range.End, ToEndOfToken);
+  }
+
+  void addFuncDecl(const FuncDecl *FD) {
+    assert(FD->hasBody() && "Must have a body");
+
+    auto *Params = FD->getParameters();
+
+    // First chunk: start -> the parameter to remove (if any)
+    SourceLoc LeftEndLoc = Params->getLParenLoc().getAdvancedLoc(1);
+    if (TopHandler.Index - 1 >= 0) {
+      LeftEndLoc = Lexer::getLocForEndOfToken(
+          SM, Params->get(TopHandler.Index - 1)->getEndLoc());
+    }
+    addRange(FD->getStartLoc(), LeftEndLoc);
+
+    // Second chunk: end of the parameter to remove -> right parenthesis
+    SourceLoc MidStartLoc = LeftEndLoc;
+    SourceLoc MidEndLoc = Params->getRParenLoc().getAdvancedLoc(1);
+    if (TopHandler.isValid()) {
+      if ((size_t)(TopHandler.Index + 1) < Params->size()) {
+        MidStartLoc = Params->get(TopHandler.Index + 1)->getStartLoc();
+      } else {
+        MidStartLoc = Params->getRParenLoc();
+      }
+    }
+    addRange(MidStartLoc, MidEndLoc);
+
+    // Third chunk: add in async and throws if necessary
+    OS << " async";
+    if (FD->hasThrows() || TopHandler.HasError)
+      // TODO: Add throws if converting a function and it has a converted call
+      //       without a do/catch
+      OS << " " << tok::kw_throws;
+
+    // Fourth chunk: if no parent handler (ie. not adding an async
+    // alternative), the rest of the decl. Otherwise, add in the new return
+    // type
+    if (!TopHandler.isValid()) {
+      SourceLoc RightStartLoc = MidEndLoc;
+      if (FD->hasThrows()) {
+        RightStartLoc = Lexer::getLocForEndOfToken(SM, FD->getThrowsLoc());
+      }
+      SourceLoc RightEndLoc = FD->getBody()->getLBraceLoc();
```

Comment:
Similar to above, it might be worth checking if the LBraceLoc is guaranteed to be valid here with a func decl that's missing a body.

Reply:
`findFunction` has a check for there being a body, so it should never get in here if there isn't one. Though I'm not sure what happens if either left/right is missing, I'll double check that. Might be worth adding an assert here as well regardless though. Thanks :)

Reply:
Yep, this case works fine, I've added an assert anyway